scipy/stats/_multivariate.py

```diff
@@ -385,11 +393,39 @@ def __call__(self, mean=None, cov=1, allow_singular=False, seed=None):
                                           allow_singular=allow_singular,
                                           seed=seed)
 
-    def _process_parameters(self, dim, mean, cov):
+    def _process_parameters(self, mean, cov, allow_singular=True):
         """
         Infer dimensionality from mean or covariance matrix, ensure that
         mean and covariance are full vector resp. matrix.
         """
+        if isinstance(cov, _covariance.Covariance):
+            return self._process_parameters_Covariance(mean, cov)
+        else:
+            # This is separate because `_process_parameters_psd` was written
+            # before Covariance objects were introduced. The intent is to
+            # maintain (almost) exactly the same behavior when `cov` is an
+            # array, but we don't want to have to branch the logic in every
+            # method depending on whether `cov` is a Covariance object or an
+            # array, so if `cov` is an array we use the original input
+            # validation code and wrap the old `_PSD` object in a light
+            # wrapper to match the Covariance interface.
```

Comment:
```suggestion
            # Before `Covariance` classes were introduced,
            # `multivariate_normal` accepted plain arrays as `cov` and used the
            # following input validation. To avoid disturbing the behavior of
            # `multivariate_normal` when plain arrays are used, we use the
            # original input validation here.
```